src/app/g3w-ol/controls/querybypolygoncontrol.js

```diff
@@ -87,15 +87,16 @@ proto.listenLayersVisibilityChange = function() {
   this.unwatches.forEach(unwatch => unwatch());
   this.unwatches.splice(0);
   this.layers.forEach(layer => {
-    const {state} = layer;
     this.unwatches.push(
-      VM.$watch(() =>  state.visible, visible => {
+      VM.$watch(() =>  layer.state.visible, visible => {
```

Comment:
double check if coded in this way can lead to problems additional issues

That means, before there was an object destructuring + function clousure for the `layer.state` property?